packages/alchemy/src/react/hooks/useSignerStatus.ts

```diff
@@ -0,0 +1,25 @@
+"use client";
+
+import { useSyncExternalStore } from "react";
+import { getSignerStatus } from "../../config/actions/getSignerStatus.js";
+import { watchSignerStatus } from "../../config/actions/watchSignerStatus.js";
+import { AlchemySignerStatus } from "../../signer/types.js";
+import { useAlchemyAccountContext } from "../context.js";
+
+const serverStatus = {
+  status: AlchemySignerStatus.INITIALIZING,
+  isInitializing: true,
+  isAuthenticating: false,
+  isConnected: false,
+  isDisconnected: false,
+};
+
+export const useSignerStatus = () => {
+  const { config } = useAlchemyAccountContext();
+
+  return useSyncExternalStore(
+    watchSignerStatus(config),
+    () => getSignerStatus(config),
+    () => serverStatus
```

Comment:
this is requirement of the `useSyncExternalStore` hook. the value should be static

Reply:
sorry, i mean instead of null - per your comment

Reply:
oh! yea I had it as `null` initially, but this was causing some flashes of content or a react error because on the hydrated state wasn't matching the server state